src/lib.rs

```diff
@@ -409,16 +517,35 @@ impl Navigator {
     pub fn read_adc_all(&mut self) -> ADCData {
         ADCData {
             channel: [
-                self.read_adc(adc_Channel::SingleA0),
-                self.read_adc(adc_Channel::SingleA1),
-                self.read_adc(adc_Channel::SingleA2),
-                self.read_adc(adc_Channel::SingleA3),
+                self.read_adc(AdcChannel::Ch0),
+                self.read_adc(AdcChannel::Ch1),
+                self.read_adc(AdcChannel::Ch2),
+                self.read_adc(AdcChannel::Ch3),
             ],
         }
     }
 
-    pub fn read_adc(&mut self, channel: adc_Channel) -> i16 {
-        block!(self.adc.read(channel)).unwrap()
+    /// Reads the ADC based on ADS1115 of [`Navigator`].
+    ///
+    /// Measurements in \[V\]
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use navigator_rs::{AdcChannel, Navigator};
+    /// use std::thread::sleep;
+    /// use std::time::Duration;
+    ///
+    /// let mut nav = Navigator::new();
+    /// nav.init();
+    ///
+    /// loop {
+    ///     println!("ADC channel 1 value: {} [V]", nav.read_adc(AdcChannel::Ch1));
+    ///     sleep(Duration::from_millis(1000));
+    /// }
+    pub fn read_adc(&mut self, channel: AdcChannel) -> f32 {
+        let conversion_volts: f32 = 0.000_125;
```

Comment:
can you explain why this weird value? Might be easier for us to understand if we divide instead of multiply

Reply:
it's the LSB value for ±4.096 V Scale.
It's using 4.096, but it's limited to max 3.3V which is IC's Vdd.

[FullScale Register](https://www.ti.com/lit/ds/symlink/ads1115.pdf#page=17)

Reply:
can you put the math explicitly on the code ? Or add a comment to make it clear ?
Right now is not possible to understand just looking at the code.

Reply:
This is explained with a single comment now. 

I am planning to create a PR to the ads crate. I am considering adding a new method called read_v, which will takes the current scale and perform the conversion.